### PR TITLE
Ben

### PR DESCRIPTION
**Pull request by BenTho-Uni**
Merged both of your contributions to my branch with slightly different arguments. Added one unit test per course assignment. Added unit3 prep. This okay with you guys?

### CODE REVIEW

**Review by Snertson — APPROVED** [review]
sieht soweit gut aus. Müssen nur noch unsere To Do abarbeiten/besprechen :+1:

**Review by BauerMarco — COMMENTED** [review]
Looks good!
A couple of things though:
1. I think you kicked out my plot function for the non-fft autocorr plot. (I could deal with that, but I want to make sure, just in case you moved it somewhere else.)
2. If we give the filepath_out as an argument, we might as well make that the full path (with name.pdf), instead of just the directory, where we still determine the filename in the function.
3. Does the unit test work now? Because you still left a comment in there, telling us it does not work!

**Comment by BenTho-Uni** [comment]
> 
> 
> Looks good!
> A couple of things though:
> 
>     1. I think you kicked out my plot function for the non-fft autocorr plot. (I could deal with that, but I want to make sure, just in case you moved it somewhere else.)
> 
>     2. If we give the filepath_out as an argument, we might as well make that the full path (with name.pdf), instead of just the directory, where we still determine the filename in the function.
> 
>     3. Does the unit test work now? Because you still left a comment in there, telling us it does not work!

Sorry, didn't refresh this page and couldn't see your comment before merging xD
1. Yep, my mistake. I overlooked that function, thanks!
2. yes, that would also be possible. Something we should decide. It's just one short change each. 
3. My bad, again. The unit test with self.assertListEqual works, but not the one with filecmp.cmp; 

I'll commit the changes in a second
2.

**Pull request merged**
